src/Faulkner/state_manager.py

```diff
@@ -55,11 +56,28 @@ def move_to_next_state(self):
             return
 
         # if yaw error below tolerance threshold, straight ahead, else fix yaw
-        if math.fabs(self.state_vars.err_yaw) <= self.params.yaw_precision:
-            self.set_action_state(2) # go straight ahead
-        else:
+        if math.fabs(self.state_vars.err_yaw) > self.params.yaw_precision and not self.is_oscillating():
             self.set_action_state(1) # fix yaw
+        else:
+            self.set_action_state(2) # go straight ahead
     
+    def is_oscillating(self):
+        history_window = self.history.get()
+        fix_yaw_count = 0
+        last_ang_z = 1
+        num_oscillations = 0
+        for iteration in history_window[::-1]: #reverse iterator
+            if iteration.action_state == 1: # is fix yaw iteration
+                fix_yaw_count += 1
+                if fix_yaw_count > 1 and iteration.ang_z * last_ang_z < 0:
+                    num_oscillations += 1
+                last_ang_z = iteration.ang_z
+
+        is_oscillating = num_oscillations >= self.params.yaw_oscillation_tolerance
```

Comment:
This variable and the function have the same name; change one or the other.

Reply:
done